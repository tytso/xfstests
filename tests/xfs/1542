#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2022 Oracle.  All Rights Reserved.
#
# FS QA Test No. 1542
#
# Populate a XFS filesystem and fuzz every rtrefcountbt key/ptr field.
# Use xfs_scrub to fix the corruption.

. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_scrub dangerous_online_repair realtime

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy
. ./common/reflink

# real QA test starts here
_supported_fs xfs
_require_realtime
_require_scratch_reflink
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

path="$(_scratch_xfs_find_rgbtree_height 'refcnt' 2)" || \
	_fail "could not find two-level rtrefcountbt"
inode_ver=$(_scratch_xfs_get_metadata_field "core.version" "path -m $path")

echo "Fuzz rtrefcountbt keyptrs"
_scratch_xfs_fuzz_metadata '(rtrefcbt)' 'online' "path -m $path" >> $seqres.full
echo "Done fuzzing rtrefcountbt keyptrs"

# success, all done
status=0
exit
