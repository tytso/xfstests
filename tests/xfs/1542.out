QA output created by 1542
Format and populate
Fuzz rtrefcountbt keyptrs
u3.rtrefcbt.level = ones: mount failed (32).
u3.rtrefcbt.level = firstbit: mount failed (32).
u3.rtrefcbt.level = middlebit: mount failed (32).
u3.rtrefcbt.level = add: mount failed (32).
u3.rtrefcbt.level = sub: mount failed (32).
u3.rtrefcbt.numrecs = ones: mount failed (32).
u3.rtrefcbt.numrecs = firstbit: mount failed (32).
u3.rtrefcbt.numrecs = middlebit: mount failed (32).
u3.rtrefcbt.numrecs = add: mount failed (32).
u3.rtrefcbt.numrecs = sub: mount failed (32).
Done fuzzing rtrefcountbt keyptrs
